BOM upload after clone of project might get stuck/fail

### Current Behavior

I do something like this:

1. clone project via  API (includeAuditHistory=true, includeComponents=true)
2. check token (/events...) until processing=false
3. upload BOM
4. check token (/events...) until processing=false
5. check results
6. delete project

Quite often it happens that the upload of the BOM fails with something like 
```
Insert of object "org.dependencytrack.model.ProjectMetadata@6cb89346" using statement "INSERT INTO "PROJECT_METADATA" ("AUTHORS","PROJECT_ID","SUPPLIER") VALUES (?,?,?)" failed : ERROR: duplicate key value violates unique constraint "PROJECT_METADATA_PROJECT_ID_IDX"
  Detail: Key ("PROJECT_ID")=(215) already exists.
```
<details>
  <summary>Stacktrace</summary>

```
javax.jdo.JDODataStoreException: Insert of object "org.dependencytrack.model.ProjectMetadata@27963218" using statement "INSERT INTO "PROJECT_METADATA" ("AUTHORS","PROJECT_ID","SUPPLIER") VALUES (?,?,?)" failed : ERROR: duplicate key value violates unique constraint "PROJECT_METADATA_PROJECT_ID_IDX"
  Detail: Key ("PROJECT_ID")=(269) already exists.
	at org.datanucleus.api.jdo.JDOAdapter.getJDOExceptionForNucleusException(JDOAdapter.java:605)
	at org.datanucleus.api.jdo.JDOPersistenceManager.flush(JDOPersistenceManager.java:2057)
	at org.dependencytrack.tasks.BomUploadProcessingTaskV2.processProject(BomUploadProcessingTaskV2.java:369)
	at org.dependencytrack.tasks.BomUploadProcessingTaskV2.lambda$processBom$0(BomUploadProcessingTaskV2.java:297)
	at org.dependencytrack.persistence.QueryManager.lambda$runInTransaction$0(QueryManager.java:1433)
	at org.dependencytrack.persistence.QueryManager.runInTransaction(QueryManager.java:1464)
	at org.dependencytrack.persistence.QueryManager.runInTransaction(QueryManager.java:1432)
	at org.dependencytrack.tasks.BomUploadProcessingTaskV2.processBom(BomUploadProcessingTaskV2.java:296)
	at org.dependencytrack.tasks.BomUploadProcessingTaskV2.processEvent(BomUploadProcessingTaskV2.java:187)
	at org.dependencytrack.tasks.BomUploadProcessingTaskV2.inform(BomUploadProcessingTaskV2.java:162)
	at alpine.event.framework.BaseEventService.lambda$publish$0(BaseEventService.java:110)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "PROJECT_METADATA_PROJECT_ID_IDX"
  Detail: Key ("PROJECT_ID")=(269) already exists.
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2725)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2412)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:371)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:502)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:419)
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:194)
	at org.postgresql.jdbc.PgPreparedStatement.executeUpdate(PgPreparedStatement.java:155)
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)
	at org.datanucleus.store.rdbms.SQLController.doExecuteStatementUpdate(SQLController.java:463)
	at org.datanucleus.store.rdbms.SQLController.executeStatementUpdateDeferRowCountCheckForBatching(SQLController.java:413)
	at org.datanucleus.store.rdbms.request.InsertRequest.execute(InsertRequest.java:532)
	at org.datanucleus.store.rdbms.RDBMSPersistenceHandler.insertObjectInTable(RDBMSPersistenceHandler.java:235)
	at org.datanucleus.store.rdbms.RDBMSPersistenceHandler.insertObject(RDBMSPersistenceHandler.java:211)
	at org.datanucleus.state.StateManagerImpl.internalMakePersistent(StateManagerImpl.java:4614)
	at org.datanucleus.state.StateManagerImpl.flush(StateManagerImpl.java:5848)
	at org.datanucleus.flush.FlushOrdered.execute(FlushOrdered.java:96)
	at org.datanucleus.ExecutionContextImpl.flushInternal(ExecutionContextImpl.java:4050)
	at org.datanucleus.ExecutionContextImpl.flush(ExecutionContextImpl.java:3996)
	at org.datanucleus.api.jdo.JDOPersistenceManager.flush(JDOPersistenceManager.java:2040)
	... 12 common frames omitted
```
</details>

This is logged by `BomUploadProcessingTaskV2`. The error message is somehow similar to https://github.com/DependencyTrack/dependency-track/issues/3324

 It also seems to be dependent on the project. Some work without issues and some mostly fail but I cannot really spot a difference (apart from components/suppressions)

You may wonder why I clone the project, just to delete it right after :) Im working on an integration of dtrack analysis in our pipelines. For each PR I want to analyse policy violations&vulns on the basis of the audit history of the "main" version. I think there was once an issue for this feature but I cannnot find it anymore.

### Steps to Reproduce

1. clone project via 
2. check for token (/events...) until processing=false
3. upload BOM

### Expected Behavior

No error

### Dependency-Track Version

4.11.4

### Dependency-Track Distribution

Container Image

### Database Server

PostgreSQL

### Database Server Version

15.7

### Browser

Google Chrome

### Checklist

- [X] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [X] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this defect was already reported

--- a/src/main/java/org/dependencytrack/common/MdcKeys.java
+++ b/src/main/java/org/dependencytrack/common/MdcKeys.java
@@ -30,6 +30,7 @@ public final class MdcKeys {
     public static final String MDC_BOM_SPEC_VERSION = "bomSpecVersion";
     public static final String MDC_BOM_UPLOAD_TOKEN = "bomUploadToken";
     public static final String MDC_BOM_VERSION = "bomVersion";
+    public static final String MDC_EVENT_TOKEN = "eventToken";
     public static final String MDC_PROJECT_NAME = "projectName";
     public static final String MDC_PROJECT_UUID = "projectUuid";
     public static final String MDC_PROJECT_VERSION = "projectVersion";

--- a/src/main/java/org/dependencytrack/persistence/ProjectQueryManager.java
+++ b/src/main/java/org/dependencytrack/persistence/ProjectQueryManager.java
@@ -578,151 +578,160 @@ final class ProjectQueryManager extends QueryManager implements IQueryManager {
     }
 
     @Override
-    public Project clone(UUID from, String newVersion, boolean includeTags, boolean includeProperties,
-                         boolean includeComponents, boolean includeServices, boolean includeAuditHistory,
-                         boolean includeACL, boolean includePolicyViolations) {
-        final Project source = getObjectByUuid(Project.class, from, Project.FetchGroup.ALL.name());
-        if (source == null) {
-            LOGGER.warn("Project with UUID %s was supposed to be cloned, but it does not exist anymore".formatted(from));
-            return null;
-        }
-        if (doesProjectExist(source.getName(), newVersion)) {
-            // Project cloning is an asynchronous process. When receiving the clone request, we already perform
-            // this check. It is possible though that a project with the new version is created synchronously
-            // between the clone event being dispatched, and it being processed.
-            LOGGER.warn("Project %s was supposed to be cloned to version %s, but that version already exists"
-                    .formatted(source, newVersion));
-            return null;
-        }
-        Project project = new Project();
-        project.setAuthor(source.getAuthor());
-        project.setManufacturer(source.getManufacturer());
-        project.setSupplier(source.getSupplier());
-        project.setPublisher(source.getPublisher());
-        project.setGroup(source.getGroup());
-        project.setName(source.getName());
-        project.setDescription(source.getDescription());
-        project.setVersion(newVersion);
-        project.setClassifier(source.getClassifier());
-        project.setActive(source.isActive());
-        project.setCpe(source.getCpe());
-        project.setPurl(source.getPurl());
-        project.setSwidTagId(source.getSwidTagId());
-        if (includeComponents && includeServices) {
-            project.setDirectDependencies(source.getDirectDependencies());
-        }
-        project.setParent(source.getParent());
-        project = persist(project);
-
-        if (source.getMetadata() != null) {
-            final var metadata = new ProjectMetadata();
-            metadata.setProject(project);
-            metadata.setAuthors(source.getMetadata().getAuthors());
-            metadata.setSupplier(source.getMetadata().getSupplier());
-            persist(metadata);
-        }
-
-        if (includeTags) {
-            for (final Tag tag: source.getTags()) {
-                tag.getProjects().add(project);
-                persist(tag);
+    public Project clone(
+            final UUID from,
+            final String newVersion,
+            final boolean includeTags,
+            final boolean includeProperties,
+            final boolean includeComponents,
+            final boolean includeServices,
+            final boolean includeAuditHistory,
+            final boolean includeACL,
+            final boolean includePolicyViolations
+    ) {
+        final Project clonedProject = callInTransaction(() -> {
+            final Project source = getObjectByUuid(Project.class, from, Project.FetchGroup.ALL.name());
+            if (source == null) {
+                LOGGER.warn("Project was supposed to be cloned, but it does not exist anymore");
+                return null;
             }
-        }
-
-        if (includeProperties && source.getProperties() != null) {
-            for (final ProjectProperty sourceProperty: source.getProperties()) {
-                final ProjectProperty property = new ProjectProperty();
-                property.setProject(project);
-                property.setPropertyType(sourceProperty.getPropertyType());
-                property.setGroupName(sourceProperty.getGroupName());
-                property.setPropertyName(sourceProperty.getPropertyName());
-                property.setPropertyValue(sourceProperty.getPropertyValue());
-                property.setDescription(sourceProperty.getDescription());
-                persist(property);
+            if (doesProjectExist(source.getName(), newVersion)) {
+                // Project cloning is an asynchronous process. When receiving the clone request, we already perform
+                // this check. It is possible though that a project with the new version is created synchronously
+                // between the clone event being dispatched, and it being processed.
+                LOGGER.warn("Project was supposed to be cloned to version %s, but that version already exists".formatted(newVersion));
+                return null;
             }
-        }
+            Project project = new Project();
+            project.setAuthor(source.getAuthor());
+            project.setManufacturer(source.getManufacturer());
+            project.setSupplier(source.getSupplier());
+            project.setPublisher(source.getPublisher());
+            project.setGroup(source.getGroup());
+            project.setName(source.getName());
+            project.setDescription(source.getDescription());
+            project.setVersion(newVersion);
+            project.setClassifier(source.getClassifier());
+            project.setActive(source.isActive());
+            project.setCpe(source.getCpe());
+            project.setPurl(source.getPurl());
+            project.setSwidTagId(source.getSwidTagId());
+            if (includeComponents && includeServices) {
+                project.setDirectDependencies(source.getDirectDependencies());
+            }
+            project.setParent(source.getParent());
+            project = persist(project);
 
-        final Map<Long, Component> clonedComponents = new HashMap<>();
-        if (includeComponents) {
-            final List<Component> sourceComponents = getAllComponents(source);
-            if (sourceComponents != null) {
-                for (final Component sourceComponent: sourceComponents) {
-                    final Component clonedComponent = cloneComponent(sourceComponent, project, false);
-                    // Add vulnerabilties and finding attribution from the source component to the cloned component
-                    for (Vulnerability vuln: sourceComponent.getVulnerabilities()) {
-                        final FindingAttribution sourceAttribution = this.getFindingAttribution(vuln, sourceComponent);
-                        this.addVulnerability(vuln, clonedComponent, sourceAttribution.getAnalyzerIdentity(), sourceAttribution.getAlternateIdentifier(), sourceAttribution.getReferenceUrl(), sourceAttribution.getAttributedOn());
-                    }
-                    clonedComponents.put(sourceComponent.getId(), clonedComponent);
+            if (source.getMetadata() != null) {
+                final var metadata = new ProjectMetadata();
+                metadata.setProject(project);
+                metadata.setAuthors(source.getMetadata().getAuthors());
+                metadata.setSupplier(source.getMetadata().getSupplier());
+                persist(metadata);
+            }
+
+            if (includeTags) {
+                for (final Tag tag : source.getTags()) {
+                    tag.getProjects().add(project);
+                    persist(tag);
                 }
             }
-        }
 
-        if (includeServices) {
-            final List<ServiceComponent> sourceServices = getAllServiceComponents(source);
-            if (sourceServices != null) {
-                for (final ServiceComponent sourceService : sourceServices) {
-                    cloneServiceComponent(sourceService, project, false);
+            if (includeProperties && source.getProperties() != null) {
+                for (final ProjectProperty sourceProperty : source.getProperties()) {
+                    final ProjectProperty property = new ProjectProperty();
+                    property.setProject(project);
+                    property.setPropertyType(sourceProperty.getPropertyType());
+                    property.setGroupName(sourceProperty.getGroupName());
+                    property.setPropertyName(sourceProperty.getPropertyName());
+                    property.setPropertyValue(sourceProperty.getPropertyValue());
+                    property.setDescription(sourceProperty.getDescription());
+                    persist(property);
                 }
             }
-        }
 
-        if (includeAuditHistory && includeComponents) {
-            final List<Analysis> analyses = super.getAnalyses(source);
-            if (analyses != null) {
-                for (final Analysis sourceAnalysis: analyses) {
-                    Analysis analysis = new Analysis();
-                    analysis.setAnalysisState(sourceAnalysis.getAnalysisState());
-                    final Component clonedComponent = clonedComponents.get(sourceAnalysis.getComponent().getId());
-                    if (clonedComponent == null) {
-                        break;
+            final Map<Long, Component> clonedComponents = new HashMap<>();
+            if (includeComponents) {
+                final List<Component> sourceComponents = getAllComponents(source);
+                if (sourceComponents != null) {
+                    for (final Component sourceComponent : sourceComponents) {
+                        final Component clonedComponent = cloneComponent(sourceComponent, project, false);
+                        // Add vulnerabilties and finding attribution from the source component to the cloned component
+                        for (Vulnerability vuln : sourceComponent.getVulnerabilities()) {
+                            final FindingAttribution sourceAttribution = this.getFindingAttribution(vuln, sourceComponent);
+                            this.addVulnerability(vuln, clonedComponent, sourceAttribution.getAnalyzerIdentity(), sourceAttribution.getAlternateIdentifier(), sourceAttribution.getReferenceUrl(), sourceAttribution.getAttributedOn());
+                        }
+                        clonedComponents.put(sourceComponent.getId(), clonedComponent);
                     }
-                    analysis.setComponent(clonedComponent);
-                    analysis.setVulnerability(sourceAnalysis.getVulnerability());
-                    analysis.setSuppressed(sourceAnalysis.isSuppressed());
-                    analysis.setAnalysisResponse(sourceAnalysis.getAnalysisResponse());
-                    analysis.setAnalysisJustification(sourceAnalysis.getAnalysisJustification());
-                    analysis.setAnalysisState(sourceAnalysis.getAnalysisState());
-                    analysis.setAnalysisDetails(sourceAnalysis.getAnalysisDetails());
-                    analysis = persist(analysis);
-                    if (sourceAnalysis.getAnalysisComments() != null) {
-                        for (final AnalysisComment sourceComment: sourceAnalysis.getAnalysisComments()) {
-                            final AnalysisComment analysisComment = new AnalysisComment();
-                            analysisComment.setAnalysis(analysis);
-                            analysisComment.setTimestamp(sourceComment.getTimestamp());
-                            analysisComment.setComment(sourceComment.getComment());
-                            analysisComment.setCommenter(sourceComment.getCommenter());
-                            persist(analysisComment);
+                }
+            }
+
+            if (includeServices) {
+                final List<ServiceComponent> sourceServices = getAllServiceComponents(source);
+                if (sourceServices != null) {
+                    for (final ServiceComponent sourceService : sourceServices) {
+                        cloneServiceComponent(sourceService, project, false);
+                    }
+                }
+            }
+
+            if (includeAuditHistory && includeComponents) {
+                final List<Analysis> analyses = super.getAnalyses(source);
+                if (analyses != null) {
+                    for (final Analysis sourceAnalysis : analyses) {
+                        Analysis analysis = new Analysis();
+                        analysis.setAnalysisState(sourceAnalysis.getAnalysisState());
+                        final Component clonedComponent = clonedComponents.get(sourceAnalysis.getComponent().getId());
+                        if (clonedComponent == null) {
+                            break;
+                        }
+                        analysis.setComponent(clonedComponent);
+                        analysis.setVulnerability(sourceAnalysis.getVulnerability());
+                        analysis.setSuppressed(sourceAnalysis.isSuppressed());
+                        analysis.setAnalysisResponse(sourceAnalysis.getAnalysisResponse());
+                        analysis.setAnalysisJustification(sourceAnalysis.getAnalysisJustification());
+                        analysis.setAnalysisState(sourceAnalysis.getAnalysisState());
+                        analysis.setAnalysisDetails(sourceAnalysis.getAnalysisDetails());
+                        analysis = persist(analysis);
+                        if (sourceAnalysis.getAnalysisComments() != null) {
+                            for (final AnalysisComment sourceComment : sourceAnalysis.getAnalysisComments()) {
+                                final AnalysisComment analysisComment = new AnalysisComment();
+                                analysisComment.setAnalysis(analysis);
+                                analysisComment.setTimestamp(sourceComment.getTimestamp());
+                                analysisComment.setComment(sourceComment.getComment());
+                                analysisComment.setCommenter(sourceComment.getCommenter());
+                                persist(analysisComment);
+                            }
                         }
                     }
                 }
             }
-        }
 
-        if (includeACL) {
-            List<Team> accessTeams = source.getAccessTeams();
-            if (!CollectionUtils.isEmpty(accessTeams)) {
-                project.setAccessTeams(new ArrayList<>(accessTeams));
+            if (includeACL) {
+                List<Team> accessTeams = source.getAccessTeams();
+                if (!CollectionUtils.isEmpty(accessTeams)) {
+                    project.setAccessTeams(new ArrayList<>(accessTeams));
+                }
             }
-        }
 
-     
-       if(includeComponents && includePolicyViolations){
-            final List<PolicyViolation> sourcePolicyViolations = getAllPolicyViolations(source);
-            if(sourcePolicyViolations != null){
-                for(final PolicyViolation policyViolation: sourcePolicyViolations){
-                final Component destinationComponent = clonedComponents.get(policyViolation.getComponent().getId());
-                final PolicyViolation clonedPolicyViolation = clonePolicyViolation(policyViolation, destinationComponent);
-                persist(clonedPolicyViolation);
-                }   
+
+            if (includeComponents && includePolicyViolations) {
+                final List<PolicyViolation> sourcePolicyViolations = getAllPolicyViolations(source);
+                if (sourcePolicyViolations != null) {
+                    for (final PolicyViolation policyViolation : sourcePolicyViolations) {
+                        final Component destinationComponent = clonedComponents.get(policyViolation.getComponent().getId());
+                        final PolicyViolation clonedPolicyViolation = clonePolicyViolation(policyViolation, destinationComponent);
+                        persist(clonedPolicyViolation);
+                    }
+                }
             }
-       }
-        
 
-        project = getObjectById(Project.class, project.getId());
-        Event.dispatch(new IndexEvent(IndexEvent.Action.CREATE, project));
+            return project;
+        });
+
+        Event.dispatch(new IndexEvent(IndexEvent.Action.CREATE, clonedProject));
         commitSearchIndex(true, Project.class);
-        return project;
+        return clonedProject;
     }
 
     /**

--- a/src/main/java/org/dependencytrack/tasks/CloneProjectTask.java
+++ b/src/main/java/org/dependencytrack/tasks/CloneProjectTask.java
@@ -25,8 +25,12 @@ import org.dependencytrack.event.CloneProjectEvent;
 import org.dependencytrack.model.Project;
 import org.dependencytrack.persistence.QueryManager;
 import org.dependencytrack.resources.v1.vo.CloneProjectRequest;
+import org.slf4j.MDC;
 
 import java.util.UUID;
+
+import static org.dependencytrack.common.MdcKeys.MDC_EVENT_TOKEN;
+import static org.dependencytrack.common.MdcKeys.MDC_PROJECT_UUID;
 
 public class CloneProjectTask implements Subscriber {
 
@@ -36,15 +40,26 @@ public class CloneProjectTask implements Subscriber {
      * {@inheritDoc}
      */
     public void inform(final Event e) {
-        if (e instanceof CloneProjectEvent) {
-            final CloneProjectEvent event = (CloneProjectEvent)e;
+        if (e instanceof final CloneProjectEvent event) {
             final CloneProjectRequest request = event.getRequest();
-            LOGGER.info("Cloning project: " + request.getProject());
-            try (QueryManager qm = new QueryManager()) {
-                final Project project = qm.clone(UUID.fromString(request.getProject()),
-                        request.getVersion(), request.includeTags(), request.includeProperties(),
-                        request.includeComponents(), request.includeServices(), request.includeAuditHistory(), request.includeACL(), request.includePolicyViolations());
-                LOGGER.info("Cloned project: " + request.getProject() + " to " + project.getUuid());
+            try (var ignoredMdcProjectUuid = MDC.putCloseable(MDC_PROJECT_UUID, request.getProject());
+                 var ignoredMdcEventToken = MDC.putCloseable(MDC_EVENT_TOKEN, event.getChainIdentifier().toString());
+                 final var qm = new QueryManager()) {
+                LOGGER.info("Cloning project for version %s".formatted(request.getVersion()));
+                final Project project = qm.clone(
+                        UUID.fromString(request.getProject()),
+                        request.getVersion(),
+                        request.includeTags(),
+                        request.includeProperties(),
+                        request.includeComponents(),
+                        request.includeServices(),
+                        request.includeAuditHistory(),
+                        request.includeACL(),
+                        request.includePolicyViolations()
+                );
+                LOGGER.info("Cloned project for version %s into project %s".formatted(project.getVersion(), project.getUuid()));
+            } catch (RuntimeException ex) {
+                LOGGER.error("Failed to clone project", ex);
             }
         }
     }

--- a/src/test/java/org/dependencytrack/tasks/BomUploadProcessingTaskTest.java
+++ b/src/test/java/org/dependencytrack/tasks/BomUploadProcessingTaskTest.java
@@ -1216,6 +1216,83 @@ public class BomUploadProcessingTaskTest extends PersistenceCapableTest {
         });
     }
 
+    @Test
+    public void informIssue3981Test() {
+        final var project = new Project();
+        project.setName("acme-license-app");
+        project.setVersion("1.2.3");
+        qm.persist(project);
+
+        byte[] bomBytes = """
+                {
+                  "bomFormat": "CycloneDX",
+                  "specVersion": "1.6",
+                  "serialNumber": "urn:uuid:3e671687-395b-41f5-a30f-a58921a69b80",
+                  "version": 1,
+                  "metadata": {
+                    "authors": [
+                      {
+                        "name": "foo",
+                        "email": "foo@example.com"
+                      }
+                    ]
+                  },
+                  "components": [
+                    {
+                      "type": "library",
+                      "name": "acme-lib-x"
+                    }
+                  ]
+                }
+                """.getBytes(StandardCharsets.UTF_8);
+
+        var bomUploadEvent = new BomUploadEvent(qm.detach(Project.class, project.getId()), bomBytes);
+        new BomUploadProcessingTask().inform(bomUploadEvent);
+        awaitBomProcessedNotification(bomUploadEvent);
+        NOTIFICATIONS.clear();
+
+        final Project clonedProject = qm.clone(project.getUuid(), "3.2.1", true, true, true, true, true, true, true);
+
+        bomBytes = """
+                {
+                  "bomFormat": "CycloneDX",
+                  "specVersion": "1.6",
+                  "serialNumber": "urn:uuid:3e671687-395b-41f5-a30f-a58921a69b80",
+                  "version": 1,
+                  "metadata": {
+                    "authors": [
+                      {
+                        "name": "bar",
+                        "email": "bar@example.com"
+                      }
+                    ]
+                  },
+                  "components": [
+                    {
+                      "type": "library",
+                      "name": "acme-lib-x"
+                    }
+                  ]
+                }
+                """.getBytes(StandardCharsets.UTF_8);
+
+        bomUploadEvent = new BomUploadEvent(qm.detach(Project.class, clonedProject.getId()), bomBytes);
+        new BomUploadProcessingTask().inform(bomUploadEvent);
+        awaitBomProcessedNotification(bomUploadEvent);
+
+        qm.getPersistenceManager().evictAll();
+
+        assertThat(project.getMetadata().getAuthors()).satisfiesExactly(author -> {
+            assertThat(author.getName()).isEqualTo("foo");
+            assertThat(author.getEmail()).isEqualTo("foo@example.com");
+        });
+
+        assertThat(clonedProject.getMetadata().getAuthors()).satisfiesExactly(author -> {
+            assertThat(author.getName()).isEqualTo("bar");
+            assertThat(author.getEmail()).isEqualTo("bar@example.com");
+        });
+    }
+
     private void awaitBomProcessedNotification(final BomUploadEvent bomUploadEvent) {
         try {
             await("BOM Processed Notification")

